<launch>

    <!-- BEGIN COMMON PARAMETERS -->

    <!-- depending on application increase this number -->
    <param name="~num_markers" type="int" value="100" />
    <param name="~camera_frame" type="string" value="left_optical_frame" />
    <param name="~target_frame" type="string" value="left_optical_frame" />
    <param name="~scale_factor" type="double" value="0.0105445269" />
    <!--param name="thin_markers" type="bool" value="false" /-->
    <!-- turn off x output to decrease load -->
    <param name="~no_x_output" type="bool" value="false" />
    <param name="~frame_rate" type="int" value="10" />
    <!-- proportional to size of markers -->
    <param name="~block_size" type="int" value="31" />
    <!-- increase if poor contrast -->
    <param name="~offset" type="double" value="15.0" />

    <!-- END COMMMON PARAMETERS -->

    <!-- first stereo node -->
    <node pkg="marker_detect" name="l_marker_detector_stream" type="marker_detector_node" output="screen" args="output_x">
        <param name="pub_topic" type="string" value="l_marker_detector" />
        <param name="xy_pub_topic" type="string" value="l_marker_xys" />
    	<remap from="image_rect" to="/left/image_rect" />
        <remap from="camera_info" to="/left/camera_info" />
    </node>

    <!-- second stereo node -->
    <node pkg="marker_detect" name="r_marker_detector_stream" type="marker_detector_node" output="screen" args="output_x">
        <param name="pub_topic" type="string" value="r_marker_detector" />
        <param name="xy_pub_topic" type="string" value="r_marker_xys" />
    	<remap from="image_rect" to="/right/image_rect" />
        <remap from="camera_info" to="/right/camera_info" />
    </node>

    <!-- combined node -->
    <node pkg="stereo_marker_detect" name="marker_detector_stereo" type="stereo_marker_detector.py" output="screen">
        <param name="l_marker_topic" type="string" value="l_marker_detector" />
        <param name="r_marker_topic" type="string" value="r_marker_detector" />
        <param name="l_marker_xy_topic" type="string" value="l_marker_xys" />
        <param name="r_marker_xy_topic" type="string" value="r_marker_xys" />
        <param name="l_cam_info_topic" type="string" value="/left/camera_info" />
        <param name="r_cam_info_topic" type="string" value="/right/camera_info" />
        <param name="output_topic" type="string" value="stereo_marker_detector" />
        <param name="output_frame" type="string" value="left_optical_frame" />
    </node>
    
</launch>
